build-system/compile/internal-version.js

```diff
@@ -26,40 +26,75 @@ const argv = minimist(process.argv.slice(2), {
 function getVersion() {
   if (argv.version_override) {
     const version = String(argv.version_override);
-    // #27579: What are allowed version strings...?
     if (!/^\d{13}$/.test(version)) {
       throw new Error('--version_override only accepts a 13-digit version');
     }
     return version;
-  } else {
-    // Generate a consistent version number by using the commit* time of the
-    // latest commit on the active branch as the twelve digits. The last,
-    // thirteenth digit defaults to 0, but can be changed by setting the
-    // --custom_version_mark flag to a different value. This should rarely be
-    // used by AMP release engineers.
-    //
-    // e.g., the version number of a clean (no uncommited changes) tree that was
-    // commited on August 1, 2018 at 14:31:11 EDT would be `1808011831110`
-    // (notice that due to timezone shift, the hour value changes from EDT's 14
-    // to UTC's 18. The last digit is the default value of 0 as
-    // --custom_version_mark was not set.)
-    //
-    // *Commit time is different from author time! Commit time is the time that
-    // the PR was merged into master; author time is when the author ran the
-    // "git commit" command.
-    const lastCommitFormattedTime = gitCommitFormattedTime();
-    let lastDigit = 0;
-    if (argv.custom_version_mark) {
-      lastDigit = parseInt(argv.custom_version_mark, 10);
-      if (isNaN(lastDigit) || lastDigit < 0 || lastDigit > 9) {
-        throw new Error(
-          `--custom_version_mark is set to ${argv.custom_version_mark}, ` +
-            'expected value between 0 and 9!'
-        );
-      }
+  }
+  // Version numbers are determined using the following algorithm:
+  // - Count the number (<X>) of cherry-picked releases on this branch that came
+  //   from the `master` branch, until reaching `master` or the first commit
+  //   that was added directly on this branch (if the current commit is on
+  //   `master`'s commit history, or only contains new commits that are not
+  //   cherry-picked from `master`, then <X> is 0).
+  //   - If <X> > 9 then cap it at 9 (or throw an error for `--strict_build`s).
+  // - Find the commit (<C>) before the last cherry-picked commit from the
+  //   `master` branch (if the current branch is `master`, or otherwise in
+  //   `master`'s commit history, then the current commit is <C>).
+  // - Find the commit time of <C> (<C>.time). Note that commit time might be
+  //   different from author time! e.g., commit time might be the time that a PR
+  //   was merged into `master`, or a commit was cherry-picked onto the brabnch;
+  //   author time is when the original author of the commit ran the
+  //   "git commit" command.
+  // - The version number is <C>.time.format("YYmmDDHHMMSS", "UTC") + <X> (the
+  //   pseudo-code assumes standard `.strftime` formatting).
```

Comment:
Would it be easier/clearer to speak relative to the RTV of the main release that's being cherry-picked, rather than the commit time?

Reply:
This is what this PR does :D
RTV is made our of the commit time (see #16631)